src/work_pool.c

```diff
@@ -208,32 +207,46 @@ work_pool_thread(void *arg)
 		clock_gettime(CLOCK_REALTIME_FAST, &ts);
 		timespec_addms(&ts, pool->timeout_ms);
 
+		wpt->wakeup = false;
+
 		/* Note: the mutex is the pool _head,
 		 * but the condition is per worker,
 		 * making the signal efficient!
 		 */
 		rc = pthread_cond_timedwait(&wpt->pqcond, &pool->pqh.qmutex,
 					    &ts);
-		if (!wpt->work) {
```

Comment:
The assumption here is that this thread was woken up if wpt->work is true. If not, it has to be just a timeout and we don't want to depend on the return code rc here. Why is this NOT working? Don't we set wpt->work before waking up a thread with this patch? Oh ok, you are just waking up the thread and let it get the work from the work queue!

Reply:
yes, I am not assigning the work, thread can keep on picking up the work without going to wait. This is similar to what we do for send queues.